sabnzbd/deobfuscate_filenames.py

```diff
@@ -0,0 +1,263 @@
+#!/usr/bin/python3 -OO
+# Copyright 2007-2020 The SABnzbd-Team <team@sabnzbd.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+"""
+
+Deobfuscation post-processing script:
+
+Will check in the completed job folder if maybe there are par2 files,
+for example "rename.par2", and use those to rename the files.
+If there is no "rename.par2" available, it will rename the largest
+file to the job-name in the queue.
+
+NOTES:
+ 1) To use this script you need Python installed on your system and
+    select "Add to path" during its installation. Select this folder in
+    Config > Folders > Scripts Folder and select this script for each job
+    you want it used for, or link it to a category in Config > Categories.
+ 2) Beware that files on the 'Cleanup List' are removed before
+    scripts are called and if any of them happen to be required by
+    the found par2 file, it will fail.
+ 3) If there are multiple larger (>40MB) files, then the script will not
+    rename anything, since it could be a multi-pack.
+ 4) If you want to modify this script, make sure to copy it out
+    of this directory, or it will be overwritten when SABnzbd is updated.
+ 5) Feedback or bugs in this script can be reported in on our forum:
+    https://forums.sabnzbd.org/viewforum.php?f=9
+
+
+Improved by P1nGu1n
+
+"""
+
+import os
+import sys
+import time
+import fnmatch
+import struct
+import hashlib
+import re
+import math
+import logging
+from sabnzbd.filesystem import get_unique_filename  # this makes this module SAB specific ...
+
+
+# Files to exclude and minimal file size for renaming
+EXCLUDED_FILE_EXTS = (".vob", ".rar", ".par2")
```

Comment:
Same as in `sorting.py`, so should be moved to `constants.py`.

Reply:
Ah:
https://github.com/sabnzbd/sabnzbd/blob/develop/sabnzbd/sorting.py#L44-L45

So 

```
# Do not rename .vob files as they are usually DVD's
EXCLUDED_FILE_EXTS = (".vob", ".bin")
```
So, yes no .VOB files. But .bin ... is that DVD specific?

And in deobfuscate I don't want to rename RAR and PAR2 because they are remains of a (faulty) download.

What do you think?

Reply:
Oh I didn't check well enough. Indeed these are different and don't need to be merged!